pkg/hooks/hooks_test.go

```diff
@@ -77,3 +79,62 @@ func testManifest() pods.Manifest {
 	builder.SetID(podId)
 	return builder.GetManifest()
 }
+
+func TestHookRunWithTimeout(t *testing.T) {
+	type sleepTest struct {
+		sleep      int
+		timeout    int
+		shouldPass bool
+	}
+
+	var sleepTests = []sleepTest{
+		{1, 2, true},
+		{2, 1, false},
```

Comment:
i'd settle for tests that short, definitely easier that way. although you'd have to use a command other than `sleep` and then we're getting into race condition territory